subsys/dfu/src/dfu_target_mcuboot.c

```diff
@@ -75,24 +108,46 @@ int dfu_target_mcuboot_init(size_t file_size)
 {
 	int err = flash_img_init(&flash_img);
 
+	if (err != 0) {
+		LOG_ERR("flash_img_init error %d", err);
+		return err;
+	}
+
 	if (file_size > PM_MCUBOOT_SECONDARY_SIZE) {
 		LOG_ERR("Requested file too big to fit in flash");
 		return -EFBIG;
 	}
 
-	if (err != 0) {
-		LOG_ERR("flash_img_init error %d", err);
-		return err;
-	}
+	if (IS_ENABLED(CONFIG_SETTINGS)) {
+		static struct settings_handler sh = {
+			.name = MODULE,
+			.h_set = settings_set,
+		};
 
-	offset = 0;
+		err = settings_subsys_init();
```

Comment:
I'm thinking maybe this should be moved outside the module and be handled in the application using it? For me, that makes more sense but it would require the user to initialize the settings subsys before doing a dfu.

Reply:
It depends on the implementation of `settings_subsys_init()`. If it's idempotent, IMO it should be called where the settings subsys is used. If it's not idempotent, calling it here might interfere with other users of the settings subsys, so it should be left up to the app.

Reply:
I looked now. The implementation is idempotent, though the docs don't mention it explicitly.

I also see that it's not a convention for `settings_subsys_init()` to be called from within other subsystems, so if you want to remove it, I won't object.

Ideally, we should make a PR upstream adding it to the function docs, since it is a bit tedious when enabling a subsystem in Kconfig also means you must add the init function to your main.

Alternatively, the init could be called via SYS_INIT(). I don't really know Zephyr's conventions for things like this :).

Reply:
Yup also found that it was idempotent added a comment about it. 
I think only drivers are put into SYS_INIT while subsystems should explicitly initialized I don't know if this is convention or just my assumption.

Reply:
I'll leave this as unresolved so that I can remember to make an upstream commit to the documentation.